### PR TITLE
Bug 1940159: Fix Routers destroy with BYO

### PR DESCRIPTION
**Pull request by MaysaMacedo**
When using your own Network for the Machines
connected to a tagged Router created by CNO
on an installation with Kuryr, the Subnet of
that Network is not tagged and consequently
not identified by installer when destroying
the Router. This commit fixes the issue by
ensuring all Subnetes connected to a Router
created by Installer or CNO are removed.

### CODE REVIEW

**Comment by mandre** [comment]
We're in feature freeze - you'll need an associated BZ to merge anything before master opens again for 4.9 development work.

File: pkg/destroy/openstack/openstack.go [hunk 1]
```diff
@@ -686,25 +691,35 @@ func removeRouterInterfaces(client *gophercloud.ServiceClient, filter Filter, ro
 	return len(allPorts) == len(customInterfaces), nil
 }
 
-func getRouterByPort(client *gophercloud.ServiceClient, allPorts []ports.Port) (string, error) {
+func isClusterRouter(clusterTag string, tags []string, logger logrus.FieldLogger) bool {
```
[the remark this answers is not in the document]
Comment (MaysaMacedo) [reply pkg/destroy/openstack/openstack.go]: ah, leftover. Thanks for spotting!

**Pull request merged**